src/umzug.ts

```diff
@@ -377,6 +370,18 @@ export class Umzug<Ctx extends object = object> extends emittery<UmzugEvents<Ctx
 				toWrite.push([filepath, '']);
 			}
 
+			if (!options.allowConfusingOrdering) {
+				const confusinglyOrdered = existing.find(e => {
+					const existingPath = e.path;
+					return existingPath && toWrite.some(([newPath]) => existingPath >= newPath);
+				});
```

Comment:
I think this should only compare against `filepath`, not the whole `toWrite` array. `toWrite` tends to be ancillary files after the first entry, like down migrations, which don't necessarily get picked up by `glob`. It might just be luck that the test cases don't trip up on this.

```suggestion
				const confusinglyOrdered = existing.find(e => e.path && filepath >= e.path);
```